.github/workflows/scripts/release-branch.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash -ex
+#
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# --- Configuration ---
+RELEASE_SERIES="$1" # e.g., v0.85.x
+PREPARE_RELEASE_COMMIT_HASH="$2" # Optional: Specific commit hash for "Prepare release"
+UPSTREAM_REMOTE_NAME=${UPSTREAM_REMOTE_NAME:-"upstream"} # Your upstream remote name for open-telemetry/opentelemetry-collector
+MAIN_BRANCH_NAME=${MAIN_BRANCH_NAME:-"main"}
+LOCAL_MAIN_BRANCH_NAME=${LOCAL_MAIN_BRANCH_NAME:-"${MAIN_BRANCH_NAME}"}
+# These variables are only used if git user.name and git user.email are not configured
+GIT_CONFIG_USER_NAME=${GIT_CONFIG_USER_NAME:-"opentelemetrybot"}
+GIT_CONFIG_USER_EMAIL=${GIT_CONFIG_USER_EMAIL:-"107717825+opentelemetrybot@users.noreply.github.com"}
+
+# --- Helper Functions ---
+check_command_success() {
+  if [[ $? -ne 0 ]]; then
```

Comment:
Would  [`set -e`](https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#set--e) be enough to avoid this helper? 

Also, instead of calling this directly [you can use `trap`](https://stackoverflow.com/a/185900)